conda_smithy/lint_recipe.py

```diff
@@ -449,6 +449,34 @@ def lintify(meta, recipe_dir=None, conda_forge=False):
                 )
                 continue
 
+    # 23: non noarch builds shouldn't use version constraints on python
+    host_reqs = requirements_section.get("host", [])
+    run_reqs = requirements_section.get("run", [])
+    if build_section.get("noarch") is None:
+        filtered_host_reqs = [
+            req
+            for req in host_reqs
+            if req == "python" or req.startswith("python ")
+        ]
+        filtered_run_reqs = [
+            req
+            for req in run_reqs
+            if req == "python" or req.startswith("python ")
+        ]
+        if filtered_host_reqs and not filtered_run_reqs:
+            lints.append(
+                "If python is a host requirement, it should be a run requirement."
+            )
+        for reqs in [filtered_host_reqs, filtered_run_reqs]:
+            if "python" in reqs:
+                continue
+            for req in reqs:
+                constraint = req.split(" ", 1)[1]
+                if constraint.startswith(">") or constraint.startswith("<"):
+                    lints.append(
+                        "Non noarch: python packages should have a python requirement without any version constraints."
```

Comment:
This should be implemented once we drop 2.7, right? B/c it is fair game to constraint <3 or >=3 at the moment. Or am I missing something?

Reply:
If it's not a noarch package, the constraints should go in `skip: True`, otherwise you get broken packages.

Reply:
+1 to that. We need to fix the lint to accept that particular skip too. I guess you already tackle too.